src/v/cluster/tm_stm.cc

```diff
@@ -680,51 +799,92 @@ ss::future<tm_stm::op_status> tm_stm::add_group(
     co_return tm_stm::op_status::success;
 }
 
+bool tm_stm::check_tx_membership(const kafka::transactional_id& tx_id) {
```

Comment:
this the only place where we talk about hosted txs as membership, it's better not to avoid introducing new term and pick something like `tm_stm::hosts(const kafka::transactional_id&)` as a name